packages/core/src/build-utils/build/utils/loader.ts

```diff
@@ -167,6 +167,82 @@ export function mapEachRules<T extends Plugin.BuildRuleSetRule>(
   });
 }
 
+export function changeBuiltinLoader<T extends Plugin.BuildRuleSetRule>(
+  rules: T[],
+  loaderName: string,
+  appendRules: (rule: T, index: number) => T,
+): T[] {
+  return rules.map((rule) => {
+    if (!rule) return rule;
+
+    // Check the 'loader' field
+    if (
+      rule.loader &&
+      typeof rule.loader === 'string' &&
+      rule.loader === loaderName
+    ) {
+      const _rule = {
+        ...rule,
+        use: [
+          {
+            loader: rule.loader,
+            options: rule.options,
+          },
+        ],
+        loader: undefined,
+        options: undefined,
+      };
+      return appendRules(_rule, 0);
+    }
+
+    if (rule.use) {
+      if (Array.isArray(rule.use)) {
+        const _index = findIndex(
+          rule.use,
+          (_r) =>
+            (typeof _r === 'object' &&
```

Comment:
It is recommended to extract some small helper functions to increase code readability

Reply:
> It is recommended to extract some small helper functions to increase code readability

fixed